[Bug] Typescript 6 compatibility broken

## 🐛 Bug Description

`src/boot/maz-ui.ts(6,8): error TS2882: Cannot find module or type declarations for side-effect import of 'maz-ui/styles'.`

## 📦 Affected Package(s)

<!-- Check all packages that are affected by this bug -->

- [x] `maz-ui` (main component library)
- [ ] `@maz-ui/nuxt` (Nuxt module)
- [ ] `@maz-ui/icons` (icon library)
- [x] `@maz-ui/themes` (theme system)
- [x] `@maz-ui/translations` (i18n)
- [ ] `@maz-ui/forms` (forms)
- [ ] `@maz-ui/utils` (utilities)
- [ ] `@maz-ui/cli` (CLI tools)
- [ ] `@maz-ui/eslint-config` (ESLint config)
- [ ] `@maz-ui/mcp` (MCP of maz-ui)
- [ ] `@maz-ui/node` (Node utilities)
- [ ] `docs` (documentation)
- [ ] Other:

## 🔧 Component/Feature

<!-- If applicable, specify the component or feature name -->

## 🔄 Steps to Reproduce

<!-- Steps to reproduce the behavior -->

Use the IntelliJ IDE, or run the website (quasar dev) or build the website (quasar build).

## ✅ Expected Behavior

No error.

## ❌ Actual Behavior

`src/boot/maz-ui.ts(6,8): error TS2882: Cannot find module or type declarations for side-effect import of 'maz-ui/styles'.`

## 🖼️ Screenshots/Code

<!-- If applicable, add screenshots or code snippets to help explain your problem -->

qusar.config.ts
```typescript
boot: ['maz-ui'],
```

src/boot/maz-ui.ts
```typescript
import { defineBoot } from '#q-app/wrappers'
import { MazUi } from 'maz-ui/plugins/maz-ui'
import { mazUi } from '@maz-ui/themes/presets'
import { de, en } from '@maz-ui/translations'

import 'maz-ui/styles'

export default defineBoot(({ app }) => {
  app.use(MazUi, {
    theme: {
      preset: mazUi,
    },
    translations: {
      locale: 'de',
      fallbackLocale: 'en',
      messages: {
        de,
        en,
      },
    },
  })
})
```

## 🌍 Environment

**Framework:**

- [ ] Vue
- [ ] Nuxt
- [x] Other: Quasar

**Package Version:**

- Maz-UI: 4.9.0
- Vue: 3.5.31
- Quasar: 2.19.1
- Node: 24.12.0
- pnpm: 10.33.0

**Browser:**

- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge
- [ ] Mobile browsers
- [x] Other: Chromium

**Operating System:**

- [ ] macOS
- [ ] Windows
- [x] Linux
- [ ] Mobile (iOS/Android)
- [ ] Other:

## 📋 Additional Context

Upgrade Typescript to 6.0.2

## 🔍 Possible Solution

Downgrade to Typescript 5.9.3

## 📝 Checklist

<!-- Please check the following before submitting -->

- [x] I have searched for existing issues
- [x] I have tested this on the latest version
- [x] I have provided all necessary information
- [x] I have included code examples/screenshots where applicable

--- a/packages/lib/build/ViteCompileStyles.ts
+++ b/packages/lib/build/ViteCompileStyles.ts
@@ -55,6 +55,13 @@ export function ViteCompileStyles(): Plugin {
         await compileScss()
 
         logger.success('[CompileStyles] ✅ scss compiled')
+
+        // Write type declaration stubs for TypeScript 6 compatibility (TS2882)
+        // Side-effect CSS imports require a `types` field in package.json subpath exports
+        writeFileSync(resolve(cssDir, 'main.css.d.ts'), 'export {}\n')
+        writeFileSync(resolve(cssDir, 'aos.css.d.ts'), 'export {}\n')
+
+        logger.success('[CompileStyles] ✅ type declarations written')
       }
       catch (error) {
         logger.error('[CompileStyles] 🔴 error while compiling styles', error)

--- a/packages/lib/package.json
+++ b/packages/lib/package.json
@@ -109,8 +109,14 @@
       "module": "./dist/tailwindcss/*.js",
       "default": "./dist/tailwindcss/*.js"
     },
-    "./styles": "./dist/css/main.css",
-    "./aos-styles": "./dist/css/aos.css",
+    "./styles": {
+      "types": "./dist/css/main.css.d.ts",
+      "default": "./dist/css/main.css"
+    },
+    "./aos-styles": {
+      "types": "./dist/css/aos.css.d.ts",
+      "default": "./dist/css/aos.css"
+    },
     "./*": "./*"
   },
   "main": "./dist/index.js",

